app/controllers/categories_controller.rb

```diff
@@ -14,45 +15,33 @@ def show; end
   private
 
   def init_values
-    @active_more_btn = false
-    @page_number = 1
+    @present = CategoriesPresenter.new(params: params)
     load_books_with_category(params) if params[:id]
-    load_books_without_category(params) unless params[:id]
-    show_more_button?(params)
+    load_books_without_category unless params[:id]
+    @active_more_btn = show_more_button?(params)
   end
 
-  def load_books_without_category(params)
+  def load_books_without_category
     @category = nil
-    init_page_number(params)
-    init_filter(params)
-    @books = Book.by_filter(@filter, @page_number)
+    @books = Book.by_filter(@present.filter, @present.page_number)
   end
 
   def load_books_with_category(params)
     @category = Category.find(params[:id])
     category_id = params[:id]
-    init_page_number(params)
-    init_filter(params)
     @books = Book.where(category_id: category_id)
-                 .by_filter(@filter, @page_number)
-  end
-
-  def init_page_number(params)
-    @page_number = params[:page].present? ? params[:page].to_i + 1 : 1
-  end
-
-  def init_filter(params)
-    @filter = params[:filter].present? ? params[:filter] : Book::DEFAULT_FILTER
+                 .by_filter(@present.filter, @present.page_number)
   end
 
   def show_more_button?(params)
-    return if @books.to_a.size.zero?
+    return false unless @books.exists?
+
     if params[:id]
-      if @books.to_a.size != qunatity_book_in_category(@category.id)
-        @active_more_btn = true
-      end
+      return true if @books.to_a.size != qunatity_book_in_category(@category.id)
```

Comment:
I think you don't need to call `to_a` here

Reply:
I called to_a because it's not realy book model after fetch
![2018-08-14 3 28 25](https://user-images.githubusercontent.com/9571192/44117758-e5bd204a-a01c-11e8-8c82-17f34c4307f5.png)
![2018-08-14 2 53 56](https://user-images.githubusercontent.com/9571192/44117867-2a1f3dcc-a01d-11e8-8c0b-7a9688df293a.png)

Reply:
You can re-write the query to this:
```
Book.left_outer_joins(:positions).group('books.id').order('COALESCE(SUM(positions.quantity), 0) desc')
```
A `.count` called on this AR relation will return a hash instead of integer.
You can use this gem to get an integer: https://github.com/mrbrdo/active_record_group_count
It's not critical in your app because you have only 50 books. But it's good to know why `count` is better: https://work.stevegrossi.com/2015/04/25/how-to-count-with-activerecord/